runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -298,6 +300,19 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
       {
       vmemParams.pageSize = largeCodePageSize;
       vmemParams.pageFlags = config.largeCodePageFlags();
+
+      // Recalculate the page size if using large pages
+      UDATA *pageSizes = j9vmem_supported_page_sizes();
+      if(vmemParams.pageSize > ccTotalSizeB)
+         {
+         for (UDATA pageIndex = 0; 0 != pageSizes[pageIndex]; ++pageIndex)
+            {
+            if(pageSizes[pageIndex] <= ccTotalSizeB)
+               vmemParams.pageSize = pageSizes[pageIndex];
+            }
+         if (config.verboseCodeCache())
+            TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE,"Warning:Using page size %zu instead of large page size %zu", vmemParams.pageSize, largeCodePageSize);
```

Comment:
`%zu` is for printing size_t. From what I know `size_t` and `uintptr_t` have the same size, so the code above should work. However, to be pedantic `uintptr_t` values need to be printed with `PRIuPTR`. I found that to be very ugly. A compromise would be to cast the values to be printed to `size_t` and continue to use `%zu`.